cdc/entry/schema_storage.go

```diff
@@ -226,21 +246,103 @@ func (s *schemaStorageImpl) HandleDDLJob(job *timodel.Job) error {
 	return nil
 }
 
+// AllPhysicalTables returns the table IDs of all tables and partition tables.
+func (s *schemaStorage) AllPhysicalTables(ctx context.Context, ts model.Ts) ([]model.TableID, error) {
+	// NOTE: it's better to pre-allocate the vector. However, in the current implementation
```

Comment:
You can add a field named `lastTableCount` to `schemaStorage` to record the number of table IDs when `AllPhysicalTables` was last called. Since this method may be called frequently in certain situations, it can help improve efficiency.